src/pip/_internal/index/package_finder.py

```diff
@@ -208,15 +210,17 @@ def evaluate_link(self, link):
 
         # This should be up by the self.ok_binary check, but see issue 2700.
         if "source" not in self._formats and ext != WHEEL_EXTENSION:
-            return (False, 'No sources permitted for %s' % self.project_name)
+            return (False, 'No sources permitted for {}'.format(
+                self.project_name))
```

Comment:
I would put each tuple element on its own line. This indentation is a bit confusing. Or just put the message in a separate statement (`reason = ...` and then `return (False, reason)`).